client/express-checkout/utils/checkPaymentMethodIsAvailable.js

```diff
@@ -15,11 +15,24 @@ import { getUPEConfig } from 'wcpay/utils/checkout';
 
 export const checkPaymentMethodIsAvailable = memoize(
 	( paymentMethod, cart, resolve ) => {
-		const root = ReactDOM.createRoot(
-			document.getElementById(
-				`express-checkout-check-availability-container-${ paymentMethod }`
-			)
-		);
+		// Create the DIV container on the fly
+		const containerEl = document.createElement( 'div' );
+
+		// Ensure the element is hidden and doesn’t interfere with the page layout.
+		containerEl.style.border = 0;
+		containerEl.style.height = '0';
+		containerEl.style.margin = '0';
+		containerEl.style.overflow = 'hidden';
+		containerEl.style.padding = '0';
+		containerEl.style.position = 'absolute';
+		containerEl.style.width = '0';
+		containerEl.style.float = 'left';
+		containerEl.style.opacity = '0';
+		containerEl.style.pointerEvents = 'none';
+
+		document.querySelector( 'body' ).appendChild( containerEl );
+
+		const root = ReactDOM.createRoot( containerEl );
```

Comment:
I also tested this with only:

```js
containerEl.style.display = 'none';
```

And it worked as expected. With `display: none`, the element shouldn't have any pointer events and shouldn't take up any space.

We might not need to keep the element technically visible by setting other properties to `0`, but perhaps I could be missing something?

The current approach also LGTM, the only downside is that it's slightly more complex and the element is still "visible" in the DOM, including for assistive technologies, although temporarily.

cc @bborman22

Reply:
I guess it doesn't matter if the button doesn't end up rendering as long as we have the `onReady` event to be called with the info we need.

Reply:
> the only downside is that it's slightly more complex and the element is still "visible" in the DOM, including for assistive technologies

Great catch! I missed the `ariaHidden` not being set. I think at a minimum that should be set. Display none is certainly simpler, just wondering if there are some gotchas there as we don't use it in other places we create a hidden div? [ref1](https://github.com/Automattic/woocommerce-payments/blob/develop/client/checkout/upe-styles/index.js#L219) [ref2](https://github.com/Automattic/woocommerce-payments/blob/develop/client/checkout/woopay/connect/woopay-connect-iframe.js#L72) with the second ref using an iframe as one possible explanation.

Reply:
I believe it would depend mostly on Stripe, e.g. if it checks for any layout measurements of the root element, or if the contents are technically visible.

Both approaches seem to work though, so I don't have a strong preference, but I agree with @bborman22 we should probably set `containerEl.setAttribute( 'aria-hidden', 'true' );`.

Reply:
I've updated the PR with the suggestion. [70191f7](https://github.com/Automattic/woocommerce-payments/pull/9420/commits/70191f780d294900cbf3cc0acf9169f1aca44717) – I've tested with regular and block-based theeme. With Safari, Google Chrome, and Firefox.